File: Merge Sorted Array.py
def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        k=m+n-1
        while m>0 and n>0:
            if nums1[m-1]>nums2[n-1]:
                nums1[k]=nums1[m-1]
                m-=1
            else:
                nums1[k]=nums2[n-1]
                n-=1
            k-=1
        while n>0:
            nums1[k]=nums2[n-1]
            n,k=n-1,k-1

File: MergeSortedArray.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int gap = ceil((float)(n + m)/2);
        
        while(gap > 0) {
            int i = 0;
            int j = gap;
            while(j < (n+m)) {
                if(j < m && nums1[i] > nums1[j])
                    swap(nums1[i], nums1[j]);
                else if (j >= m && i < m && nums1[i] > nums2[j-m])
                    swap(nums1[i], nums2[j-m]);
                else if (j>=m && i >= m && nums2[i-m] > nums2[j-m])
                    swap(nums2[i-m], nums2[j-m]);
                
                j++;
                i++;
            }
            
            if(gap == 1) {
                gap = 0;
            } else {
                gap = ceil((float) gap / 2);
            }
        }
        int k = 0;
        for(int i = m; i<m+n; i++) {
            nums1[i] = nums2[k];
            k++;
        }
    }
};
